docs/runtime/types/module-struct.md

```diff
@@ -0,0 +1,241 @@
+---
+title: "Module Struct"
+---
+
+When developing runtime in Substrate, you need to declare a `Module` struct for your runtime module. This `Module` struct wraps a set of useful functions and implementations, either written by you or generated by the macros in Substrate. 
+
+There are a few code snippets in following sections to show parts of `Module` struct implementations. If you are interested in what's fully expanded code look like for your module, try using [cargo-expand](https://github.com/dtolnay/cargo-expand) by following its instructions.
+
+## The Definition of  Module Struct
+
+Here is template runtime module provided by the node-template, it contains following code along with `decl_module!` macro:
+
+```rust
+decl_module! {
+    /// The module declaration.
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Initializing events
+        fn deposit_event<T>() = default;
+        
+        // Just a dummy entry point.
+        pub fn do_something(origin, something: u32) -> Result {
+            let who = ensure_signed(origin)?; 
+            <Something<T>>::put(something);
+            Self::deposit_event(RawEvent::SomethingStored(something, who));
+            Ok(())
+        }
+    }
+}
+```
+
+Let's look at the code that shows the definition of `Module` struct after expanded. It has a type parameter `T` which bounds to the module's configuration `Trait`. [PhantomData](https://doc.rust-lang.org/beta/std/marker/struct.PhantomData.html) is used to indicate what type of data our `Module` struct is "tied" to, but not found in our struct.
```

Comment:
I would omit the `PhantomData` stuff in your explanation and your example below. It is an implementation detail, not necessarily what you want to teach them.

Reply:
Based on the discussion, the function is not being used and provide no value for now, remove it in this doc.

Reply:
Leave the PhantomData in the code example makes sure our code is syntax correct. A bit description on `PhantomData` could reduce confusion at first place. The redundancy seems valuable from this perspective.

Reply:
But this phantom data description should not be found in the middle of this random page...

I dont think correct code syntax is what we are looking for here. We want to explain to them the general structures being made, not the specific nitty gritty details

Reply:
Fair point, the `PhantomData` removed.